File: Leetcode/C++/136. Single Number/solution.cpp
// https://leetcode.com/problems/single-number/description/

#include <vector>

using namespace std;

class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        int result = 0;
        for (auto num : nums)
        {
            result ^= num;
        }
        return result;
    }
};

File: Leetcode/Python/136. Single Number/solution.py
# https://leetcode.com/problems/single-number/


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0
        for num in nums:
            # ^: XOR if both number same result is 0 else result = none zero num
            # Ex. 5 ^ 5 = 0, 5 ^ 0 = 5
            result ^= num
        return result
